### PR TITLE
Enable keepalive pinging thanks to @asutula

### PR DESCRIPTION
**Pull request by carsonfarmer**
Signed-off-by: Carson Farmer <carson.farmer@gmail.com>

This is a direct copy of the solution added here: https://github.com/textileio/textile/pull/263

### CODE REVIEW

File: threadsd/main.go [hunk 1]
```diff
@@ -118,6 +118,7 @@ func main() {
 			return true
 		}),
 		grpcweb.WithWebsockets(true),
+		grpcweb.WithWebsocketPingInterval(time.Second*5),
```
Comment (sanderpick) [inline threadsd/main.go:121]: Can you make this a flag? Should be configurable by the user

Comment (carsonfarmer) [reply threadsd/main.go:121]: Sure

File: threadsd/main.go [hunk 2]
```diff
@@ -35,6 +35,7 @@ func main() {
 	connLowWater := fs.Int("connLowWater", 100, "Low watermark of libp2p connections that'll be maintained")
 	connHighWater := fs.Int("connHighWater", 400, "High watermark of libp2p connections that'll be maintained")
 	connGracePeriod := fs.Duration("connGracePeriod", time.Second*20, "Duration a newly opened connection is given before it becomes subject to pruning")
+	keepAliveInterval := fs.Duration("keepAliveInterval", time.Second*5, "Enables websocket keepalive pinging with the specified timeout interval. Configured interval must be >= 1s.")
```
Comment (carsonfarmer) [inline threadsd/main.go:38]: Leaving this defaulting to 5 seconds, as the Listen API is much more useful when the socket is kept alive (in the case of slow updates in realistic scenarios).

File: threadsd/main.go [hunk 3]
```diff
@@ -118,6 +119,7 @@ func main() {
 			return true
 		}),
 		grpcweb.WithWebsockets(true),
+		grpcweb.WithWebsocketPingInterval(*keepAliveInterval),
```
Comment (carsonfarmer) [inline threadsd/main.go:122]: This is the same thing that the improbable-eng lib does. They pass along keepAliveInterval regardless of value, and if it is less than 1 second, it is ignored.

**Pull request merged**